src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateIO.cs

```diff
@@ -181,12 +207,234 @@ public void UpdatePackageReference(PackageSpec updatedPackageSpec, IPackageUpdat
         }
     }
 
+    /// <inheritdoc cref="IPackageUpdateIO.GetLatestVersionAsync(string, bool, ILogger, CancellationToken)"/>
+    public async Task<NuGetVersion?> GetLatestVersionAsync(
+        string packageId,
+        bool includePrerelease,
+        ILogger logger,
+        CancellationToken cancellationToken)
+    {
+        var sources = GetSourcesForPackage(packageId);
+        var lookups = new Task<NuGetVersion?>[sources.Count];
+        for (int source = 0; source < sources.Count; source++)
+        {
+            SourceRepository sourceRepository = sources[source];
+            // If package source is a local folder feed, it might not actually be async
+            lookups[source] = Task.Run(() => FindHighestPackageVersionAsync(sourceRepository, packageId, includePrerelease, logger, cancellationToken));
+        }
+
+        await Task.WhenAll(lookups);
+
+        NuGetVersion? highestVersion = null;
+        foreach (var task in lookups)
+        {
+            if (task.Result != null)
+            {
+                if (highestVersion == null || task.Result > highestVersion)
+                {
+                    highestVersion = task.Result;
+                }
+            }
+        }
+
+        return highestVersion;
+    }
+
+    /// <inheritdoc cref="IPackageUpdateIO.GetKnownVulnerabilitiesAsync(ILogger, CancellationToken)"/>
+    public async Task<IReadOnlyList<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>>> GetKnownVulnerabilitiesAsync(ILogger logger, CancellationToken cancellationToken)
+    {
+        IReadOnlyList<PackageSource>? auditSources = _sourceProvider.LoadAuditSources()?.Where(s => s.IsEnabled).ToList();
+        if (auditSources is null || auditSources.Count == 0)
+        {
+            auditSources = _enabledSources;
+        }
+
+        var tasks = new List<Task<GetVulnerabilityInfoResult?>>(auditSources.Count);
+        foreach (var auditSource in auditSources)
```

Comment:
Interesting.... This means that vulnerability information in regular sources will not be consumed if any audit sources are defined. Maybe out of scope for this PR, but we should a conversation about the user scenarios. For example, let's say I get all my vulnerability info from NuGet.org (which is configured as a source) but want to add an additional "Audit Source" for not-yet-disclosed vulnerabilities for my packages owned by my organization. Is that a realistic scenario?